php_phongo.c

```diff
@@ -527,6 +538,34 @@ static bool process_read_preference(zval *option, bson_t *mongoc_opts, zval **zr
 	return true;
 }
 
+static bool process_session(zval *option, bson_t *mongoc_opts, zval **zsession, mongoc_client_t *client TSRMLS_DC)
+{
+	const mongoc_client_session_t *client_session;
+
+	if (Z_TYPE_P(option) != IS_OBJECT || !instanceof_function(Z_OBJCE_P(option), php_phongo_session_ce TSRMLS_CC)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"session\" option to be %s, %s given", ZSTR_VAL(php_phongo_session_ce->name), zend_get_type_by_const(Z_TYPE_P(option)));
```

Comment:
This should use the newly introduce macro from your other PR to show the class names.

Reply:
Same answer as https://github.com/mongodb/mongo-php-driver/pull/707#discussion_r157181533.

This will be updated to use macros in https://github.com/mongodb/mongo-php-driver/pull/705 after that PR is merged and this is rebased.